src/display/canvas.js

```diff
@@ -1089,6 +1089,9 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
 
       if (isPatternFill) {
         ctx.save();
+        var m = this.baseTransform ||
+                this.ctx.mozCurrentTransform.slice();
+        ctx.setTransform(m[0], m[1], m[2], m[3], m[4], m[5]);
```

Comment:
So this code just performs: `if (this.baseTransform) { ctx.setTransform.apply(ctx, this.baseTransform); }`? (Now I wonder if we ever will get `this.baseTransform == null`.)

I don't think you need to perform this.ctx.mozCurrentTransform and its slicing -- it will just create unneeded objects.

Shall stroke have the same statements?